wire/README.md

```diff
@@ -600,6 +600,25 @@ func inject() *Bar {
 }
 ```
 
+### Why does Wire forbid including the same provider multiple times?
+
```

Comment:
I think we should have some guidance for the higher level question described in the issue. Not "why does Wire make this illegal", but "why you shouldn't be doing this".

The scenario is something like:

```
app uses libA
app uses libB
libA uses libC
libB uses libC
```

and the problem comes up if both libA and libB have a ProviderSet that includes providers for things from libC. The argument is that either

1. libC is an implementation detail inside libA/libB. In this scenario, libA/libB should construct libC things internally and not expose it to the application layer. No need to include libC things in the ProviderSet.
2. libC things need some kind of configuration for libA/libB to use it. In this scenario, the app needs to know about libC things, and should construct them itself. For example, it might want to pass the same libC thing to both libA and libB. Or not. Again, libA/libB should not include libC things in their ProviderSet -- the application needs to know about libC so it should include libC's providers or ProviderSet explicitly.

Maybe a question along the lines of "What providers should a library include in its ProviderSet?" could address this?

Reply:
I see. SGTM, but it sounds more like it belongs in the best practices section above. I've added more guidance there. PTAL and let me know whether I should keep this section or remove it.